test/unit/specs/components/Personalization/index.spec.js

```diff
@@ -10,185 +10,94 @@ OF ANY KIND, either express or implied. See the License for the specific languag
-
-    expect(Array.isArray(result)).toBeTrue();
-    expect(result.length).toEqual(0);
-  });
 
-  it("expects getDecisions to return empty array if the storage is empty", () => {
-    const scopes = ["Foo1", "Foo3"];
-
-    const response = {
-      getPayloadsByType() {
-        return [];
-      }
-    };
-    const personalization = createPersonalization({
-      config,
-      logger,
-      eventManager
+    personalization.lifecycle.onBeforeEvent({
+      event,
+      decisionsScopes,
+      onResponse
     });
 
-    personalization.lifecycle.onResponse({ response });
-
-    const result = personalization.commands.getDecisions.run({ scopes });
-
-    expect(Array.isArray(result)).toBeTrue();
-    expect(result.length).toEqual(0);
+    return expectAsync(deferred.promise).toBeResolvedTo({
+      decisions: SCOPES_FOO1_FOO2_DECISIONS
+    });
   });
 
-  it("expects getDecisions to return all decisions for a specific scope stored during multiple responses", () => {
-    const scopes = ["Foo5"];
-    const first = {
-      getPayloadsByType() {
-        return NO_SCOPES_DECISIONS;
-      }
-    };
-    const second = {
-      getPayloadsByType() {
-        return SAME_SCOPE_MULTIPLE_DECISIONS;
-      }
-    };
+  it("should return an array of not rendered decisions in lifecycle::onResponse for provided decision scopes and the page wide scope", () => {
+    const renderDecisionsEnabled = true;
+    const decisionsScopes = ["Foo1", "Foo3"];
+    const decisions = SCOPES_FOO1_FOO2_DECISIONS.concat(
+      PAGE_WIDE_SCOPE_DECISIONS_WITH_DOM_ACTION_SCHEMA_ITEMS
+    );
+    response.getPayloadsByType.and.returnValue(decisions);
+
+    const expectedResponse = SCOPES_FOO1_FOO2_DECISIONS.concat(
+      PAGE_WIDE_SCOPE_DECISIONS_WITHOUT_DOM_ACTION_SCHEMA_ITEMS
+    );
     const personalization = createPersonalization({
       config,
       logger,
       eventManager
     });
 
-    personalization.lifecycle.onResponse({ response: first });
-    personalization.lifecycle.onResponse({ response: second });
-
-    const result = personalization.commands.getDecisions.run({ scopes });
+    const deferred = defer();
+    const onResponse = func => {
+      const result = func({ response });
 
-    expect(Array.isArray(result)).toBeTrue();
-    expect(result.length).toEqual(2);
-    expect(result[0].id).toEqual("TNT:ABC:A4");
-    expect(result[1].id).toEqual("TNT:ABC:A5");
-  });
+      deferred.resolve(result);
+    };
 
-  it("expects getDecisions to throw an error when options is missing", () => {
-    const personalization = createPersonalization({
-      config,
-      logger,
-      eventManager
+    personalization.lifecycle.onBeforeEvent({
+      event,
+      renderDecisionsEnabled,
+      decisionsScopes,
+      onResponse
     });
 
-    expect(() =>
-      personalization.commands.getDecisions.validateOptions()
-    ).toThrow();
+    return expectAsync(deferred.promise).toBeResolvedTo({
+      decisions: expectedResponse
+    });
```

Comment:
@ninaceban as you start to add more specs for personalization, I think you'll find that breaking down `Personalization/index.js` into smaller modules that get injected and then mocking those modules will provide a simpler way to test. See the Context or Identity components for examples. I think we'll talk about this in the engineering check-in today as well.

Reply:
yes that makes sense, thank you Aaron